modules/lss-creditserweiterung/creditserweiterung.user.js

```diff
@@ -0,0 +1,220 @@
+        setCookieRegex5: /\./g,
+        setCookieRegex6: /\n.*/g,
+        earnedCreditsCookie: 'totalecredits',
+        rankCookie: 'rang',
+        creditsNeeded: [
+            0,
+            200,
+            10000,
+            100000,
+            1000000,
+            5000000,
+            50000000,
+            500000000,
+            5000000000
+        ],
+        ranks: {
+            0: 'Aspirant',
+            200: 'Brandwacht (Manschap A)',
+            10000: 'Hoofdbrandwacht (Manschap B)',
+            100000: 'Brandmeester',
+            1000000: 'Hoofdbrandmeester',
+            5000000: 'Commandeur',
+            50000000: 'Adjunct-Hoofdcommandeur',
+            500000000: 'Hoofdcommandeur',
+            5000000000: 'Erelid'
+        },
+        reloadMessage: 'Open uw profiel<br>open en de website herladen!',
+        dropdownName: 'Kredietexpansie (von KBOE2)',
+        creditsOverview: 'Overzicht statistiek',
+        coinsProtokoll: 'undefined',
+        earnedCredits: 'Totale studiepunten',
+        creditsToNextRank: 'Credits voor de volgende rang',
+        updateMessage: 'Om waarden bij te werken, moet <br> de website opnieuw laden.'
+    };
+
+const username = $("#navbar_profile_link").text().trim();
+if (document.getElementById("navigation_top")){
+    var credits = $("#navigation_top").clone().wrap('<li>').parent().html();
+}
+if (document.getElementById("coins_top")){
+    var coins = $("#coins_top").clone().wrap('<li>').parent().html();
+}
+var earnedCredits = "undefined";
+var creditsToNextRank = "undefined";
+
+function setCookies() {
+  earnedCredits = NaN;
+  rank = NaN;
+  content = $(".lightbox_iframe");
+  content = content.first()[0];
+  content = content.contentWindow.document;
+  content = content.children[0].children[1].children[2].children[0];
+  content = content.innerHTML;
+  content = content.replace(I18n.t('lssm.creditserweiterung.setCookieRegex1'), "");
+  content = content.replace(I18n.t('lssm.creditserweiterung.setCookieRegex2'), "");
+  content = content.replace(I18n.t('lssm.creditserweiterung.setCookieRegex3'), "");
+  earnedCredits = content.replace(I18n.t('lssm.creditserweiterung.setCookieRegex4'), "");
+  earnedCredits = earnedCredits.replace(I18n.t('lssm.creditserweiterung.setCookieRegex5'), "");
+  rank = content.replace(I18n.t('lssm.creditserweiterung.setCookieRegex6'), "");
+  document.cookie = I18n.t('lssm.creditserweiterung.earnedCreditsCookie') + "=" + earnedCredits + "; path=/";
+  document.cookie = I18n.t('lssm.creditserweiterung.rankCookie') + "=" + rank + "; path=/";
+  console.log("Saved Cookies!");
+}
+
+function readCookie(name) {
+    var nameEQ = name + "=";
+    var ca = document.cookie.split(';');
+    for(var i=0;i < ca.length;i++) {
+        var c = ca[i];
+        while (c.charAt(0)==' ') c = c.substring(1,c.length);
+        if (c.indexOf(nameEQ) === 0) return c.substring(nameEQ.length,c.length);
+    }
+    return null;
+}
+
+function fnct_nextRank(){
```

Comment:
ich vermute, dass dieser Funktionsname zustande kam, weil "nextRank" schon durch eine Variable belegt ist. Grundsätzlich versucht man Funktionen so zu benennen, dass daraus ersichtlich wird was sie tun. In diesem Fall wäre also z.B. getNextRank() die passende Bezeichnung.

Reply:
das ist eine gute Idee, danke dafür ;)